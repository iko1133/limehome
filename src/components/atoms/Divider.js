import React from "react";
import styled from "styled-components/native";

const Container = styled.View`
  background-color: darkgray;
  height: 1px;
  margin: 8px 0;
  width: 100%;
`;

const Divider = (props) => {
  return <Container />;
};

export default Divider;
