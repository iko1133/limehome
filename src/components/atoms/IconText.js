import React, { useContext } from "react";
import styled, { ThemeContext } from "styled-components/native";
import { MaterialIcons } from "@expo/vector-icons";

const Container = styled.View`
  align-items: center;
  flex-direction: row;
`;

const Text = styled.Text``;

/**
 * A text accompanied by icon
 * @param {string} text - the text displayed by the component
 * @param {string} iconName - the name of one of the MaterialIcons icon to be shown by the component
 * @param {object} textStyle - styling of the text
 * @param {object} style - styling of the component container
 * @param {float} iconSize - size of the icon (Defaults to 24)
 * @param {color} iconColor - color of the icon (Defaults to theme.SPECIAL_TEXT_COLOR)
 * @param {boolean} iconLeft - whether icon should be on the left or the right of the text
 */

const IconText = ({
  text,
  iconName,
  textStyle,
  style,
  iconSize = 24,
  iconColor,
  iconLeft = true,
}) => {
  const theme = useContext(ThemeContext);

  return (
    <Container style={style}>
      {iconLeft && (
        <MaterialIcons
          name={iconName}
          size={iconSize}
          color={iconColor ?? theme.SPECIAL_TEXT_COLOR}
        />
      )}
      <Text style={textStyle}>{text}</Text>
      {!iconLeft && (
        <MaterialIcons
          name={iconName}
          size={iconSize}
          color={iconColor ?? theme.SPECIAL_TEXT_COLOR}
        />
      )}
    </Container>
  );
};

export default IconText;
