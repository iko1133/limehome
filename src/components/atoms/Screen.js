import React from "react";
import { Platform } from "react-native";
import { StatusBar } from "expo-status-bar";
import styled from "styled-components/native";

const Background = styled.KeyboardAvoidingView`
  background-color: ${({ theme }) => theme.BACKGROUND_COLOR};
  height: 100%;
  justify-content: center;
`;

const InnerContainer = styled.View`
  flex: 1;
`;

/**
 * A screen component, accounting for the most page needs to be wrapped in (like KeyboardAvoidingView)
 * @param {object} children - children object to be rendered under screen
 * @param {object} style - styling of the inner container
 */

const Screen = ({ children, style }) => {
  return (
    <Background behavior={Platform.OS === "ios" ? "padding" : "height"}>
      <InnerContainer style={style}>{children}</InnerContainer>
      <StatusBar style={"dark"} />
    </Background>
  );
};

export default Screen;
