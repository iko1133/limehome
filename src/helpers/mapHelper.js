// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
