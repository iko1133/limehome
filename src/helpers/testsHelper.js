import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
