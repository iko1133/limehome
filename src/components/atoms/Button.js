import React from "react";
import styled from "styled-components/native";

import IconText from "./IconText";

const Container = styled.TouchableOpacity`
  background-color: ${({ theme }) => theme.BUTTON_BACKGROUND_COLOR};
  border-radius: 4px;
  justify-content: center;
  height: 60px;
  padding: 0 16px;
`;

const Text = styled(IconText)``;

const Button = ({ onPress, style, ...otherProps }) => {
  return (
    <Container onPress={onPress} style={style}>
      <Text {...otherProps} />
    </Container>
  );
};

export default Button;
