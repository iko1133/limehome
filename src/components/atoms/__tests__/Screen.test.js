import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import Button from "../Button";
import Divider from "../Divider";
import Screen from "../Screen";
import IconText from "../IconText";

describe("Testing Screen", () => {
  it("Renders Screen", () =>
    expect(renderer.create(WrapInThemeProvider(<Screen />))).toMatchSnapshot());

  it("Renders Screen with divider in it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <Screen>
            <Divider />
          </Screen>
        )
      )
    ).toMatchSnapshot());

  it("Renders Screen with button in it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <Screen>
            <Button text="Press" />
          </Screen>
        )
      )
    ).toMatchSnapshot());

  it("Renders Screen with button and iconText in it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <Screen>
            <Button text="Press" />
            <IconText text="Text with icon" iconName="star" />
          </Screen>
        )
      )
    ).toMatchSnapshot());
});
