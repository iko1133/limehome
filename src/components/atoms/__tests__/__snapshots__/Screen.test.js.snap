// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Screen Renders Screen 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
          "height": "100%",
          "justifyContent": "center",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;

exports[`Testing Screen Renders Screen with button and iconText in it 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
          "height": "100%",
          "justifyContent": "center",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4D6447",
          "borderRadius": 4,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text />
        <Text
          style={
            Array [
              Object {},
            ]
          }
        >
          Press
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {},
          ]
        }
      >
        Text with icon
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing Screen Renders Screen with button in it 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
          "height": "100%",
          "justifyContent": "center",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4D6447",
          "borderRadius": 4,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text />
        <Text
          style={
            Array [
              Object {},
            ]
          }
        >
          Press
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Testing Screen Renders Screen with divider in it 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#F7F1E9",
          "height": "100%",
          "justifyContent": "center",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "darkgray",
            "height": 1,
            "marginBottom": 8,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 8,
            "width": "100%",
          },
        ]
      }
    />
  </View>
</View>
`;
